ibis/expr/tests/test_geospatial.py

```diff
@@ -1,66 +1,147 @@
 """ Tests for geo spatial data types"""
+import sys
+
 import pytest
 
 import ibis
-
-
-@pytest.mark.parametrize(
-    'modifier', ['', ';4326', ';4326:geometry', ';4326:geography']
+from ibis.expr.tests.mocks import (
+    GeoMockConnectionOmniSciDB,
+    GeoMockConnectionPostGIS,
 )
-def test_geo_literals_smoke(modifier):
-    """Smoke tests for geo spatial literals"""
-    point = (0, 1)
-    ibis.literal(point, type='point{}'.format(modifier))
-
-    linestring = [(0, 1), (2, 3)]
-    ibis.literal(linestring, type='linestring{}'.format(modifier))
-
-    polygon = [
-        ((0, 0), (4, 0), (4, 4), (0, 4), (0, 0)),
-        ((1, 1), (2, 1), (2, 2), (1, 2), (1, 1)),
-    ]
-    ibis.literal(polygon, type='polygon{}'.format(modifier))
 
-    polygon1 = (
-        (0, 0), (1, 0), (0.5, 1), (0, 0)
-    )
-    polygon2 = (
-        ((35, 10), (45, 45), (15, 40), (10, 20), (35, 10)),
-        ((20, 30), (35, 35), (30, 20), (20, 30))
-    )
-    multipolygon = [polygon1, polygon2]
-    ibis.literal(multipolygon, type='multipolygon{}'.format(modifier))
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 6),
+    reason='Geo Spatial support available just for Python >= 3.6',
+)
 
+pytest.importorskip('geoalchemy2')
+pytest.importorskip('shapely')
+pytest.importorskip('geopandas')
 
-def test_geo_ops_smoke(geo_table):
-    """Smoke tests for geo spatial operations."""
-    t = geo_table
+pytest.mark.postgis
+pytest.mark.omniscidb
 
-    # alias for fields
-    point = t.geo_point
-    linestring = t.geo_linestring
-    polygon = t.geo_polygon
-    multipolygon = t.geo_multipolygon
+mock_omniscidb = GeoMockConnectionOmniSciDB()
+mock_postgis = GeoMockConnectionPostGIS()
 
-    # test ops
-    point.srid()
-    point.set_srid(4326)
-    point.x()
-    point.y()
 
-    linestring.contains(point)
-    linestring.end_point()
-    linestring.length()
-    linestring.max_distance(point)
-    linestring.point_n(1)
-    linestring.start_point()
-    linestring.x_max()
-    linestring.x_min()
-    linestring.y_max()
-    linestring.y_min()
+@pytest.mark.parametrize('backend', [mock_postgis])
+@pytest.mark.parametrize(
+    'modifier', ['', ';4326', ';4326:geometry', ';4326:geography']
+)
+@pytest.mark.parametrize(
+    'shape,value,expected',
+    [
+        ('point', (30, 10), 'POINT (30 10)'),
+        (
+            'linestring',
+            ((30, 10), (10, 30), (40, 40)),
+            'LINESTRING (30 10, 10 30, 40 40)',
+        ),
+        (
+            'polygon',
+            (
+                ((35, 10), (45, 45), (15, 40), (10, 20), (35, 10)),
+                ((20, 30), (35, 35), (30, 20), (20, 30)),
+            ),
+            (
+                'POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), '
+                + '(20 30, 35 35, 30 20, 20 30))'
```

Comment:
You don't need the `+` here because consecutive non-format string literals are concatenated within parentheses.